### ISSUE TITLE
`removeSelected` removes any pre-selected item from dropdown 

### ISSUE THREAD

**Issue by fiznool** [angular-ui/ui-select#1672]
### Bug description:

When linking the `ui-select` component to a model which points to an item in the items array, the preselected item is not displayed in the dropdown list.

### Link to minimally-working plunker that reproduces the issue:

http://plnkr.co/edit/bMlvGYc0Jfq3PX8NQcpv?p=preview

Notice that although 'Amalie' is preselected, she is not displayed in the dropdown list.

### Version of Angular, UI-Select, and Bootstrap/Select2/Selectize CSS

Angular: 1.5.0

UI-Select: 0.18.0

Bootstrap/Select2/Selectize CSS (if applicable): 3.1.1

**Comment by fiznool** [comment]
It's worth noting that this situation works fine with 0.17.1.

A temporary fix is to add a `remove-selected="false"` attribute to the `<ui-select>` tag.

**Comment by p10** [comment]
I think it's not a bug but feature ;-) https://github.com/angular-ui/ui-select/commit/3ad084f

You can switch it back to false globaly in configuration section:

```
app.config(function(uiSelectConfig) {
  uiSelectConfig.removeSelected = false;
});
```

**Comment by fiznool** [comment]
@p10 I tried this but it didn't work... I think the config block is run after the uiSelectController is created, and so [this code](https://github.com/angular-ui/ui-select/blob/7ab5f175421e8d263e4e6c5eb38a215b4c515c42/dist/select.js#L286) which caches the `removeSelected` flag is still `true`.

**Comment by user378230** [comment]
@dondi any chance you could take a look at this?

**Comment by dondi** [comment]
OK I'll investigate.

**Comment by dondi** [comment]
Quick update: I have been able to replicate this behavior and have written a unit test for it. Based on what I can replicate, the previous version (0.17.1) indeed does not remove the item from the dropdown, but the highlighted item when the dropdown is displayed for the first time does not match the pre-selected object either (initial highlighted item is always the first one).

Is that what others saw in the previous release? If this is the behavior that everyone is used to, then even if it is not entirely correct, it will be useful to know that just reverting to this will suffice. However if I’m not seeing the correct prior behavior, or if the preference is to get this working _completely_ right, then that changes the code trajectory a bit.

Please advise—thanks!

**Comment by fiznool** [comment]
From my point of view, I've upgraded from 0.14.1 where the preselected item
was correctly displayed and highlighted, my vote would be to revert to this
expected behaviour. Thanks!

**Comment by dondi** [comment]
OK, potential fix available at https://github.com/angular-ui/ui-select/pull/1690 …I toggled between `master` and this to make sure that the incorrect behavior reported here was definitely gone in the new version. All unit tests still pass, plus one more unit test was added specifically for this situation. Hope this does the trick, let me know :)

**Comment by dondi** [comment]
BTW @fiznool I was able to get this PR to implement the fully correct behavior where the pre-existing selection in the model is indeed highlighted from the get-go in the `ui-select` element. Thank you!

**Comment by mharmuth** [comment]
@p10 For me, it works when changing the value to `false` in configuration. Thanks ;)

**Comment by fiznool** [comment]
@dondi I've just tested with a manually created build using `gulp` and all works nicely, thanks!

Could you cut a new release? I'm installing using bower and so I need the `dist` folder to contain the latest code in order to update my codebase properly.

**Comment by dondi** [comment]
Greetings @fiznool —Glad to hear your tests look good! I will have to defer to @user378230 with regard to releases.

**Comment by ptata-rythmos** [comment]
I have used LimitTo filter in ui-select where i have millions of items in my dropdown. So preselected items are not visible in select-match

pre-selected item in dropdown is not populated, when reopen the window and item is not in filtered 100 items
![empty](https://user-images.githubusercontent.com/49189514/58849137-d6537e80-86a6-11e9-9bd1-9b4e33b74d2e.png)

Preselected item is populated when the item is in filtered 100 list. 
![item-in-selected-100](https://user-images.githubusercontent.com/49189514/58849142-dc495f80-86a6-11e9-812d-039773ad2d6e.png)

when no item is selected
![no-select](https://user-images.githubusercontent.com/49189514/58849144-dfdce680-86a6-11e9-9bc6-5a04adb05744.png)

I used remove-selected="false" but no use.
can somebody help me out.
